File: mobile/lib/data/models/order.dart
class OrderItemModel {
  const OrderItemModel({
    required this.id,
    required this.menuItemId,
    required this.name,
    required this.quantity,
    required this.unitPrice,
    required this.lineTotal,
  });

  final int id;
  final int menuItemId;
  final String name;
  final int quantity;
  final double unitPrice;
  final double lineTotal;

  factory OrderItemModel.fromJson(Map<String, dynamic> json) {
    return OrderItemModel(
      id: json['id'] as int,
      menuItemId: json['menu_item_id'] as int,
      name: (json['name'] ?? '') as String,
      quantity: (json['quantity'] as num?)?.toInt() ?? 0,
      unitPrice: (json['unit_price'] as num?)?.toDouble() ?? 0,
      lineTotal: (json['line_total'] as num?)?.toDouble() ?? 0,
    );
  }
}

class OrderModel {
  const OrderModel({
    required this.id,
    required this.userId,
    required this.restaurantId,
    required this.address,
    required this.paymentMethod,
    required this.status,
    required this.totalAmount,
    required this.estimatedDeliveryMinutes,
    required this.createdAt,
    required this.items,
  });

  final int id;
  final int userId;
  final int restaurantId;
  final String address;
  final String paymentMethod;
  final String status;
  final double totalAmount;
  final int estimatedDeliveryMinutes;
  final DateTime createdAt;
  final List<OrderItemModel> items;

  factory OrderModel.fromJson(Map<String, dynamic> json) {
    final rawItems = (json['items'] as List<dynamic>? ?? const []);
    return OrderModel(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      restaurantId: json['restaurant_id'] as int,
      address: (json['address'] ?? '') as String,
      paymentMethod: (json['payment_method'] ?? '') as String,
      status: (json['status'] ?? 'pending') as String,
      totalAmount: (json['total_amount'] as num?)?.toDouble() ?? 0,
      estimatedDeliveryMinutes: (json['estimated_delivery_minutes'] as num?)?.toInt() ?? 0,
      createdAt: DateTime.tryParse((json['created_at'] ?? '') as String) ?? DateTime.now(),
      items: rawItems
          .map((item) => OrderItemModel.fromJson(item as Map<String, dynamic>))
          .toList(growable: false),
    );
  }
}

File: backend/schemas/order.py
from datetime import datetime

from pydantic import BaseModel, Field


class OrderCreateItem(BaseModel):
    menu_item_id: int
    quantity: int = Field(gt=0, le=50)


class OrderCreateRequest(BaseModel):
    restaurant_id: int
    address: str = Field(min_length=5, max_length=255)
    payment_method: str = Field(min_length=2, max_length=50)
    items: list[OrderCreateItem]


class OrderItemResponse(BaseModel):
    id: int
    menu_item_id: int
    name: str
    quantity: int
    unit_price: float
    line_total: float

    model_config = {"from_attributes": True}


class OrderResponse(BaseModel):
    id: int
    user_id: int
    restaurant_id: int
    address: str
    payment_method: str
    status: str
    total_amount: float
    estimated_delivery_minutes: int
    created_at: datetime
    items: list[OrderItemResponse]

    model_config = {"from_attributes": True}
